### ISSUE TITLE
Road to v0.65 Discussion

### ISSUE THREAD

**Issue by nadiia** [react-native-community/releases#223]
_This release cut has not happened yet. We're targeting v0.65.0 RC cut to happen the week of Apr 26._

Current latest release: [v0.64.0](https://github.com/facebook/react-native/releases/tag/v0.64.0)

### Highlighted Changes
- 

### Work Required

- [ ] RC cut and publish
- [ ] Changelog draft

### Dates
We are thinking about these rough dates for this release:
- Release Candidate Cut: Week of Apr 26th
- Publish as Release Candidate: Week of Apr 26th
- Publish as Stable: End of May

**Comment by alexandrius** [comment]
Hello. Any update regarding fabric architecture?

**Comment by hramos** [comment]
@alexandrius we'll be providing an update on Fabric and the new React Native architecture soon.

**Comment by a-eid** [comment]
What's "RootTagContext_unstable" ?

**Comment by sunnylqm** [comment]
Someone please escalate this issue as it will ba a main blocker https://github.com/facebook/react-native/issues/31179

**Comment by TheSavior** [comment]
We need to get this security related fix cherry picked into RC.0 since the branch has already been cut: https://github.com/facebook/react-native/commit/33ef82ce6dfd31e1f990d438c925a0e52723e16b

**Comment by grabbou** [comment]
There's work happening in Hermes to support Catalyst https://github.com/facebook/hermes/pull/475. It's a pretty important update to Hermes, I would recommend we wait for it to land before releasing 0.65.

**Comment by Huxpro** [comment]
Hermes just made its [v0.8.0 release for RN 0.65](https://github.com/facebook/hermes/releases/tag/v0.8.0). 

- https://github.com/facebook/react-native/commit/2b67631210070de0c07b21c69fb9d0ccd9e9fcba bumped Hermes npm version to 0.8.0. 
- https://github.com/facebook/react-native/commit/c54aeccf1a8e16240e400d783dda5ec07fcf3808 is nice to have to show the Hermes version from `NewAppScreen`
- Note that i've merged https://github.com/facebook/react-native/pull/31456 to pin the hermes pod version to 0.8.0 for the 0.65 branch.

**Comment by dulinriley** [comment]
Can we pick 7579ed3f6f1a290eb9c6380bb82015f159362b02 (it just landed)? It adds support for a new feature for Hermes heap snapshots

**Comment by kelset** [comment]
if I understood correctly, unless we re-cut the branch we'll also need to cherry-pick these:
* https://github.com/facebook/react-native/commit/36b58a824ea20daa22fe7c528a3bf0ff4e6a4cb5
* https://github.com/facebook/react-native/commit/8eceee744ed9fee1eb2402f6b13bb606f6046f62

To fix the Folly/Xcode 12.5 issue properly.

**Comment by TheSavior** [comment]
That sounds right, the top post already has one of those linked, we probably need the other too.

I don't think we should recut, Hermes 0.8.0 is based on the cut and they'd need to create a new version if we recut.

**Comment by gaodeng** [comment]
Will React Native's new architecture be released in this version?

**Comment by TheSavior** [comment]
The new architecture will not be available for broad usage in this version. We are getting closer though!

**Comment by alexandrius** [comment]
> The new architecture will not be available for broad usage in this version. We are getting closer though!

How much faster is it overall? 60-70% faster?

**Comment by ACHP** [comment]
> > The new architecture will not be available for broad usage in this version. We are getting closer though!
> 
> How much faster is it overall? 60-70% faster?

@hramos already mentionned that we'll have some news about it soon here
https://github.com/react-native-community/releases/issues/223#issuecomment-832985680


@alexandrius I certainly am as impatient as you are about this new architecture but I think this thread should focus on the "road to v.65 discussion". :)

**Comment by alexandrius** [comment]
> > > The new architecture will not be available for broad usage in this version. We are getting closer though!
> > 
> > 
> > How much faster is it overall? 60-70% faster?
> 
> @hramos already mentionned that we'll have some news about it soon here
> [#223 (comment)](https://github.com/react-native-community/releases/issues/223#issuecomment-832985680)
> 
> @alexandrius I certainly am as impatient as you are about this new architecture but I think this thread should focus on the "road to v.65 discussion". :)

You are right. I am too excited.

**Comment by TheSavior** [comment]
I dug into the two Hermes related picks that @grabbou and @dulinriley flagged with the Hermes team to understand if they are required for 0.65. Neither of them fix regressions introduced in this release, but rather add additional features or support. So I think 0.65 could proceed without them so that we aren't blocked.

I've updated the OP with the requested picks that are critical, and moved some down to another section that are non critical picks.

**Comment by mikehardy** [comment]
I'm unclear on whether the branch cut or not - I see the branch in the code dropdown on github so I think that means we need a cherrypick of this one - comment copy-splatted from the 0.64.1 cherrypicks discussion

----

Issue https://github.com/facebook/react-native/issues/31245 just got a resolution!
Please please cherry pick https://github.com/facebook/react-native/commit/b0e8c1eac0a9edda12ecfa264209a8b3222afe27 and thank you :-)

**Comment by kelset** [comment]
Update on this: me and @grabbou this morning met up and cherry picked all these commits:
* https://github.com/facebook/react-native/commit/36b58a824ea20daa22fe7c528a3bf0ff4e6a4cb5
* https://github.com/facebook/react-native/commit/2b67631210070de0c07b21c69fb9d0ccd9e9fcba
* https://github.com/facebook/react-native/commit/c54aeccf1a8e16240e400d783dda5ec07fcf3808
* https://github.com/facebook/react-native/commit/33ef82ce6dfd31e1f990d438c925a0e52723e16b
* https://github.com/facebook/react-native/commit/7579ed3f6f1a290eb9c6380bb82015f159362b02
* https://github.com/facebook/react-native/commit/8eceee744ed9fee1eb2402f6b13bb606f6046f62
* https://github.com/facebook/react-native/commit/b0e8c1eac0a9edda12ecfa264209a8b3222afe27
* https://github.com/facebook/react-native/commit/b9243e00e30be057a45af6ed1916af4328c458e4

We are now doing testing; we had to hop off before finishing all our tests to hop on our work related stuff so we will have to do it async but we'll post an update after we know if we are ready for an RC0

**Comment by TheSavior** [comment]
> For the Xcode 12.5 support, it sounds like there is one remaining issue that @PeteTheHeat has flagged. It sounds like if you are using Hermes without Flipper on iOS, you might run into this error. Peter is still working on this.

Peter landed his fix for this in https://github.com/facebook/react-native/commit/b9243e00e30be057a45af6ed1916af4328c458e4

**Comment by Yonom** [comment]
Can this be cherry picked? It would unblock several issues for us

https://github.com/facebook/react-native/commit/1d924549cad75912191005c8f68dd73e15b07183

**Comment by kelset** [comment]
Update on this: after last week cherry picking and testing, we encountered a few issues. So this morning I paired with @Titozzz and we re-did the testings, and we have now a double repro of a potential 🚨 issue.

On a freshly generated RNTestProject (via the local E2E script), using [the steps](https://reactnative.dev/docs/hermes#ios) to enable Hermes for iOS, it appears that Hermes doesn't get enabled - the phrase in the header does not appear:
<img width="545" alt="Screenshot_2021-05-12_at_11 26 32 2" src="https://user-images.githubusercontent.com/16104054/118501075-46d2f900-b720-11eb-9455-7470147dc803.png">

We need to figure out what's going on but until we understand this problem the RC0 won't be released.

**Comment by Titozzz** [comment]
@kelset https://github.com/facebook/react-native/commit/cdd0256187cecd6a699443eeef6bef1e1ad92a3a needs to be cherry picked for the e2e script issue

**Comment by kelset** [comment]
thanks @Titozzz for the link.

Quick update on the issue I was describing above around Hermes on iOS: looks like it's not just a fluke/something misconfigured but a proper issue that needs to be addressed. Digging deeper it seems that it's related to this PR by @janicduplessis https://github.com/facebook/react-native/pull/30478 that "broke" a reference that is used iOS side to determine is Hermes is available - but reverting the commit locally in the 0.65-branch didn't lead to a fix so it could be that there is something else going on.

Will keep you posted

**Comment by Huxpro** [comment]
I have merged https://github.com/facebook/react-native/pull/31559 that fixed the Hermes headers issues. Please cherry pick https://github.com/facebook/react-native/commit/59abb5f378e116288cdea2f619de0c128bb0b0eb and see if RN 0.65 is unblocked.

Thanks for everyone @Titozzz @kelset @grabbou @janicduplessis @fkgozali @fabriziobertoglio1987 involved in the triage process ❤️

**Comment by kelset** [comment]
great work everyone!

Sadly now we have another issue, this time related to the CLI 😂😭😂😭😂😭 We need to fix it to get back FastRefresh, there's already a first PR in the works: https://github.com/react-native-community/cli/pull/1420

**Comment by kelset** [comment]
Another PR that we'll have to pick when it lands -> https://github.com/facebook/react-native/pull/31562

**Comment by HoSheiMa** [comment]
hm, iam ask why we dont have a good dev-tools like flutter in vscode or external program ?

**Comment by mikehardy** [comment]
@HoSheiMa I look forward to your project creating good DevTools for react-native! But not on a thread about releases. It has nothing to do with releases. Please use appropriate venues for comments.

**Comment by acoates-ms** [comment]
Not sure how widely used it is, but it looks like `localeIdentifier` was removed from `I18nManager` accidentally.  https://github.com/facebook/react-native/pull/31589 will re-add it.  -- I'd rather avoid the breaking change if possible.

**Comment by a-eid** [comment]
> Not sure how widely used it is, but it looks like `localeIdentifier` was removed from `I18nManager` accidentally.  https://github.com/facebook/react-native/pull/31589 will re-add it.  -- I'd rather avoid the breaking change if possible.

What does `localeIdentifier` do?

**Comment by kelset** [comment]
thanks @acoates-ms, here's the commit to cherry pick: https://github.com/facebook/react-native/commit/6b91ae73cdf096e15a3235ae76276f9d7fb12f7b

**Comment by kelset** [comment]
PR that we'll have to cherry pick for the Fast Refresh / Metro stuff: https://github.com/facebook/react-native/pull/31597

**Comment by kelset** [comment]
And for Android: https://github.com/facebook/react-native/pull/31593

**Comment by kelset** [comment]
Also to cherry pick: https://github.com/facebook/react-native/commit/341f061ce3ae057f3a958654e0ec3a9c4c8211ad

**Comment by dulmandakh** [comment]
also added 2 more PRs.
* https://github.com/facebook/react-native/pull/31609
* https://github.com/facebook/react-native/pull/31611

**Comment by TheSavior** [comment]
@dulmandakh, are those critically needed to get cherry picked into 0.65 to solve some regression in this release? Or could they wait to be released with 0.66?

**Comment by kelset** [comment]
I've updated the top comment with the list of commits & PRs; as soon as all the required ones are in we will do a second round of cherry picking and testing again.

For now I've placed @dulmandakh's PRs in the optional section because I don't think those two are strictly necessary but good to have. But we can move them up if they are actually mandatory for a successful release.

**Comment by danilobuerger** [comment]
@kelset the jcenter removal from @dulmandakh would be much appreciated since its sunsetted and becoming quite flaky on CI tests..

**Comment by kelset** [comment]
I agree :) but they are not technically hard blockers for having an RC0 out

**Comment by dulmandakh** [comment]
@TheSavior maven publish can wait

**Comment by dulmandakh** [comment]
RN is failing to build iOS app on M1, and the fix https://github.com/facebook/react-native/pull/31622

**Comment by dulmandakh** [comment]
Few PRs landed

- https://github.com/facebook/react-native/commit/502b81904998b800f2d960bb4a8e244988c72958
- https://github.com/facebook/react-native/commit/70da64094608f5f2e3c554ed719e9aad624e3459
- https://github.com/facebook/react-native/commit/547b4c92e4743f5b5816297f48a608ace9de6bb5

**Comment by ghoudi01** [comment]
Is there a plan for Propose a folder structure for react native and update react-native cli to generate components ??

**Comment by kelset** [comment]
@ghoudi01 this is not the correct issue for that kind of questions 🤗 if you have a proposal around that topic, I would suggest opening a conversation here https://github.com/react-native-community/discussions-and-proposals/issues?q=is%3Aissue+is%3Aopen+sort%3Aupdated-desc

**Comment by myckhel** [comment]
@kelset can you mark your reply to @ghoudi01 comment as hidden? because the reply also is off topic and hide my own too 😉  for sure 😆  No doubt about my comment 🤣 .

**Comment by kelset** [comment]
hey folks, here's an update: me and @tido64 spent the last couple hours cherry picking all the pickable stuff from the top comment plus a few more commits that were needed (you can see the full list here: https://github.com/facebook/react-native/commits/0.65-stable), plus we did a round of testing.

As part of it, and with the help of @Titozzz we have identified a problem caused by caching, which we fixed locally: https://github.com/facebook/react-native/commit/cae063798652fcf394ccf3af4645fd971ed76c19 (this will need to be backported to main via a proper PR).

After doing that we were able to do the full round of testing, and while on my machine everything was ✅ both on one of @Titozzz's machine (the other one was ✅) and on @tido64's there was an Android crash on the RNTestProject very similar to this one: https://github.com/facebook/flipper/issues/515

Since it's so inconsistent (on 2 machines it works, on two it crashes), I think it's safe to proceed with the RC0 so that more folks can get their hands on it and we can get to the bottom of that Android/Flipper-related problem; moreover, there are a couple of PRs still waiting to be merged:
* https://github.com/facebook/react-native/pull/31675
* https://github.com/facebook/hermes/pull/475

So for sure at least an RC1 will be needed.

Me and @tido64 will most likely publish RC0 tomorrow late-morning EU time.

**Comment by danilobuerger** [comment]
@kelset would it be possible to cherry pick https://github.com/facebook/react-native/commit/4d40b53c12c8ad52760c63cacde417ee876bdfb1 into rc0 as it resolves more find node issues on m1?

**Comment by kelset** [comment]
@danilobuerger @birkir given the amount of time it takes to cherry-pick + test, we will do RC0 today without those commits, but we'll try to cherry pick them for next RC

**Comment by danilobuerger** [comment]
@kelset sure I understand. Please keep in mind that people on m1 using nvm (or other node managers) will not be able to build for iOS devices without that patch. It will error out.

**Comment by tido64** [comment]
Hi folks, a quick update: We tried to publish RC0 but are hitting this issue:

```
#!/bin/bash -eo pipefail
./scripts/circleci/buck_fetch.sh
+ buck fetch ReactAndroid/src/test/java/com/facebook/react/modules
Not using buckd because watchman isn't installed.
Picked up _JAVA_OPTIONS: -XX:+UnlockExperimentalVMOptions -XX:+UseCGroupMemoryLimitForHeap
No build file at ReactAndroid/src/main/third-party/java/robolectric/4.4/BUCK when resolving target //ReactAndroid/src/main/third-party/java/robolectric/4.4:robolectric.

This error happened while trying to get dependency '//ReactAndroid/src/main/third-party/java/robolectric/4.4:robolectric' of target '//ReactAndroid/src/test/java/com/facebook/react/modules:modules'

Exited with code exit status 1
CircleCI received exit code 1
```

You can find the full log here: https://app.circleci.com/pipelines/github/facebook/react-native/9356/workflows/f434e5f7-2845-4d11-b180-45a665dba6a5/jobs/204995

We're currently investigating it, but essentially won't be able to publish until this is fixed.

**Comment by a-eid** [comment]
RC0: 
- android crashes on startup 
- iOS builds successfully on M1 machine ( using n node version manager & xcode 12.4 ).

**Comment by kelset** [comment]
Hey folks, after many tweaks and roadblocks we managed to get to the point of publishing RC0. Sadly, the problem that during testing was inconsistently happening (in 2 out of 4 machines that tested locally) about Android crash on startup seems to be happening way more.

To keep the conversation swift and reduce the overhead of comments, let's move the conversation about RC0 here: https://github.com/react-native-community/releases/issues/235
